test/tests/workspace/k3s_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package workspace
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	agent "github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	"github.com/gitpod-io/gitpod/test/tests/workspace/common"
+)
+
+func TestK3s(t *testing.T) {
+	f := features.New("k3s").
+		WithLabel("component", "workspace").
+		Assess("it should start a k3s when cgroup v2 enable", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+			defer cancel()
+
+			api := integration.NewComponentAPI(ctx, cfg.Namespace(), kubeconfig, cfg.Client())
+			t.Cleanup(func() {
+				api.Done(t)
+			})
+
+			ws, err := integration.LaunchWorkspaceDirectly(ctx, api)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func() {
+				err = integration.DeleteWorkspace(ctx, api, ws.Req.Id)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			rsa, closer, err := integration.Instrument(integration.ComponentWorkspace, "workspace", cfg.Namespace(), kubeconfig, cfg.Client(), integration.WithInstanceID(ws.Req.Id), integration.WithWorkspacekitLift(true))
+			if err != nil {
+				t.Fatalf("unexpected error instrumenting workspace: %v", err)
+			}
+			defer rsa.Close()
+			integration.DeferCloser(t, closer)
+
+			cgv2, err := common.IsCgroupV2(rsa)
+			if err != nil {
+				t.Fatalf("unexpected error checking cgroup v2: %v", err)
+			}
+
+			if !cgv2 {
+				t.Skip("This test only works for cgroup v2")
+			}
+
+			go func() {
+				var respReadyForK3s agent.ExecResponse
+				err = rsa.Call("WorkspaceAgent.Exec", &agent.ExecRequest{
+					Dir:     "/",
+					Command: "bash",
+					Args: []string{
+						"-c",
+						"curl -L https://github.com/k3s-io/k3s/releases/download/v1.23.4%2Bk3s1/k3s -o /workspace/k3s && sudo chmod +x /workspace/k3s && sudo /workspace/k3s server -d /workspace/data --flannel-backend=host-gw > /dev/null 2>&1",
+					},
+				}, &respReadyForK3s)
+			}()
+
+			var respGetPods agent.ExecResponse
+			err = rsa.Call("WorkspaceAgent.Exec", &agent.ExecRequest{
+				Dir:     "/",
+				Command: "bash",
+				Args: []string{
+					"-c",
+					"while [ ! -e /workspace/k3s ]; do sleep 1; done &&  sudo chmod +x /workspace/k3s && sleep 20 && sudo /workspace/k3s kubectl wait --for=condition=Ready nodes -l node-role.kubernetes.io/master=true --timeout 30s && sudo /workspace/k3s kubectl get nodes",
```

Comment:
This is a draft now because we need  `sleep 20`. Without it, the Text file will be busy. I plan to fix this.